app/controllers/actions/JourneyAction.scala

```diff
@@ -30,21 +28,27 @@ import scala.concurrent.{ExecutionContext, Future}
 
 class JourneyAction @Inject()(cacheService: ExportsCacheService)(
   implicit override val executionContext: ExecutionContext
-) extends ActionRefiner[AuthenticatedRequest, JourneyRequest] with SessionIdAware {
+) extends ActionRefiner[AuthenticatedRequest, JourneyRequest] {
 
-  private val logger = Logger(this.getClass())
+  private val logger = Logger(this.getClass)
 
   override def refine[A](request: AuthenticatedRequest[A]): Future[Either[Result, JourneyRequest[A]]] = {
-    implicit val hc: HeaderCarrier =
-      HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
-
-    cacheService.get(request.sessionId).map {
-      case Some(cacheModel) => Right(JourneyRequest(request, cacheModel))
-      case _                =>
-        // $COVERAGE-OFF$Trivial
-        logger.error(s"Could not obtain journey type for ${request.sessionId}")
-        // $COVERAGE-ON
-        Left(Conflict("Could not obtain information about journey type"))
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+
+    request.declarationId match {
+      case Some(id) =>
+        cacheService.get(id).map {
+          case Some(declaration) => Right(JourneyRequest(request, declaration))
+          case _ => whenMissing(request)
+        }
+      case None => Future.successful(whenMissing(request))
     }
   }
+
+  private def whenMissing[A](request: AuthenticatedRequest[_]): Either[Result, JourneyRequest[A]] = {
```

Comment:
I'll change the name of this function. Proposal: `handleMissingDeclarationId`.

Reply:
Sorted